builder/parallels/common/driver_9.go

```diff
@@ -493,34 +490,46 @@ func (d *Parallels9Driver) IPAddress(mac string, vmName string) (string, error)
 		}
 	}
 
-	if len(mostRecentIP) == 0 {
-		log.Printf("IP lease not found for MAC address %s in: %s\n", mac, d.dhcpLeaseFile)
-
-		var stdout bytes.Buffer
-		cmd := exec.Command(d.PrlctlPath, "list", vmName, "--full", "--no-header", "-o", "ip_configured")
-		cmd.Stdout = &stdout
-		if err := cmd.Run(); err != nil {
-			log.Printf("Command run failed for Virtual Machine: %s\n", vmName)
-			return "", err
-		}
+	return mostRecentIP, nil
+}
 
-		stdoutString := strings.TrimSpace(stdout.String())
-		re := regexp.MustCompile(`([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)`)
-		macMatch := re.FindAllStringSubmatch(stdoutString, 1)
+func (d *Parallels9Driver) ipWithPrlctl(vmName string) (string, error) {
+	ip := ""
+	var stdout bytes.Buffer
+	cmd := exec.Command(d.PrlctlPath, "list", vmName, "--full", "--no-header", "-o", "ip_configured")
```

Comment:
I would also add a stderr and check if there is something there, prlctl has a hit and miss when it comes to have an error but the exit code is still 0.

I know we have a regex check later on, so if the output is empty it will be catch there so it is up to you

Reply:
You are right. For a safer side, I've added & printed stderr now.